lib/cli/src/commands/run.rs

```diff
@@ -164,7 +170,23 @@ impl RunWithPathBuf {
                     ""
                 }
             )
-        })
+        });
+
+        if let Err(err) = invoke_res {
+            if let Some(coredump_path) = self.coredump_on_trap.as_ref() {
+                let source_name = self.path.to_str().unwrap_or_else(|| "unknown");
+                if let Err(coredump_err) = generate_coredump(&err, &source_name, &coredump_path) {
+                    eprintln!("warning: coredump failed to generate: {}", coredump_err);
+                    Err(err)
+                } else {
+                    Err(err.context(format!("core dumped at {}", coredump_path.display())))
+                }
+            } else {
+                Err(err)
+            }
+        } else {
+            invoke_res
+        }
```

Comment:
This can be simplified a bit using `map_err`. That way we will not need to save the result `invoke_res` and then return it

Reply:
I'm not so fan of this approach. To me, `map_err` should only be used to map/cast the error into another error, not for any I/O or application logic.

That being said, I'm fine doing that change if you think it's better.

Reply:
I tend to think the code will be a bit simpler, but I'm not strong either way.
Let's try to chime in more opinions to see what the other maintainers prefer!

Reply:
I think it should be fine for now. 

We'll be refactoring the `run.rs` module at some point to pay off technical debt, so things may get shuffled around anyway.